daemon/libnetwork/types/types.go

```diff
@@ -7,16 +7,19 @@ import (
 	"net"
 	"strconv"
 	"strings"
+	"syscall"
 
 	"github.com/ishidawataru/sctp"
 	"github.com/moby/moby/v2/errdefs"
 )
 
-// constants for the IP address type.
+// IPFamily specify IP address type(s).
+type IPFamily int
+
 const (
-	IP = iota // IPv4 and IPv6
-	IPv4
-	IPv6
+	IP   IPFamily = syscall.AF_UNSPEC // Either IPv4 or IPv6.
+	IPv4 IPFamily = syscall.AF_INET   // Internet Protocol version 4 (IPv4).
+	IPv6 IPFamily = syscall.AF_INET6  // Internet Protocol version 6 (IPv6).
```

Comment:
FWIW happy to change this back to our own values / `iota`. I thought it would be nice to draw the analogy, but if it's a concern that this implicates they _MUST_ be the same, I can change it back (or only mention the syscall ones as comment), e.g.;

```go
const (
	IP   IPFamily = iota // Either IPv4 or IPv6, similar to [syscall.AF_UNSPEC].
	IPv4                 // Internet Protocol version 4 (IPv4), similar to [syscall.AF_INET ].
	IPv6                 // Internet Protocol version 6 (IPv6), similar to [syscall.AF_INET6 ].
)
```